search/elastic_search_manager.go

```diff
@@ -0,0 +1,45 @@
+package search
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/spf13/viper"
+	elastic "gopkg.in/olivere/elastic.v6"
+)
+
+type ESManager struct {
+	esc *elastic.Client
+	url string
+}
+
+func MakeESManager(url string) *ESManager {
+	esManager := &ESManager{}
+	esManager.url = url
+	esManager.GetClient()
+	return esManager
+}
+
+func (esManager *ESManager) GetClient() (*elastic.Client, error) {
+
+	var err error
+	if esManager.esc == nil {
+
+		log.Info("Trying to set up new connection to ElasticSearch")
+		url := viper.GetString("elasticsearch.url")
+
+		esManager.esc, err = elastic.NewClient(
+			elastic.SetURL(url),
+			elastic.SetSniff(false),
+			elastic.SetHealthcheckInterval(10*time.Second),
+			elastic.SetErrorLog(log.StandardLogger()),
+		)
+	}
+	return esManager.esc, err
```

Comment:
if esManager.esc is null, make sure err is not null. Return special case of error.
This will make life easier on the client side.